client/components/recycleAndReuseComponents/AddItemMultiSelect.js

```diff
@@ -0,0 +1,75 @@
+import { Button, Flex } from '@chakra-ui/react'
+import React from 'react'
+import Select from 'react-select'
+
+const AddItemMultiSelect = ({ data, setItems, willTriggerDialog }) => {
+  const [selectedOptions, setSelectedOptions] = React.useState([])
+
+  const parseItemsIntoOptions = (categories, items) => {
+    const allOptions = []
+    categories.forEach(category => {
+      const itemsInCategory = items.filter(item => item.category === category)
+      const itemOptions = []
+      itemsInCategory.forEach(itemInCategory => {
+        const option = {
+          "label": itemInCategory.description,
+          "value": itemInCategory.id
+        }
+        itemOptions.push(option)
+      })
+
+      const categoryOptions = {
+        "label": category,
+        "options": itemOptions
+      }
+      allOptions.push(categoryOptions)
+    })
+    
+    return allOptions
+  }
```

Comment:
The [previous API endpoint](https://api.npoint.io/a8416aa207861acb363d) being used is a bit simplistic and is designed directly for the multi select. I created some [mock data](https://api.npoint.io/324727202a8ce24af12f) which is more realistic but this also means that we have to parse the data into options

Reply:
Good point @simonlau . I'll commit this change